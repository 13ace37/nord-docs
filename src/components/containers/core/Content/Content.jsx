/*
 * Copyright (C) 2018-present Arctic Ice Studio <development@arcticicestudio.com>
 * Copyright (C) 2018-present Sven Greb <development@svengreb.de>
 *
 * Project:    Nord Docs
 * Repository: https://github.com/arcticicestudio/nord-docs
 * License:    MIT
 */

import styled from "styled-components";
import { em } from "polished";

/**
 * A container that manages the responsive content width.
 * The boolean `centered` prop can be used to use a centered margin.
 *
 * @author Arctic Ice Studio <development@arcticicestudio.com>
 * @author Sven Greb <development@svengreb.de>
 * @since 0.3.0
 */
const Content = styled.div`
  width: 100%;
  margin: ${({ centered }) => centered && "0 auto"};

  ${({ theme }) => theme.media.desktop`
    max-width: ${em(theme.media.breakpoints.desktopLowerBoundary)}
  `};
`;

export default Content;
