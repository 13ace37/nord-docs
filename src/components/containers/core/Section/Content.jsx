/*
 * Copyright (C) 2018-present Arctic Ice Studio <development@arcticicestudio.com>
 * Copyright (C) 2018-present Sven Greb <development@svengreb.de>
 *
 * Project:    Nord Docs
 * Repository: https://github.com/arcticicestudio/nord-docs
 * License:    MIT
 */

import styled from "styled-components";

import CoreContent from "containers/core/Content";

/**
 * A container for content of the `Section` component.
 *
 * @author Arctic Ice Studio <development@arcticicestudio.com>
 * @author Sven Greb <development@svengreb.de>
 * @since 0.3.0
 */
const Content = styled(CoreContent)`
  padding-top: ${({ compact }) => !compact && "5em"};
  padding-bottom: ${({ compact }) => !compact && "5em"};
`;

export default Content;
