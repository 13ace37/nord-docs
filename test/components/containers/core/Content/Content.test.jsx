/*
 * Copyright (C) 2018-present Arctic Ice Studio <development@arcticicestudio.com>
 * Copyright (C) 2018-present Sven Greb <development@svengreb.de>
 *
 * Project:    Nord Docs
 * Repository: https://github.com/arcticicestudio/nord-docs
 * License:    MIT
 */

import React from "react";

import { renderWithTheme } from "nord-docs-test-utils";
import Content from "containers/core/Content";

test("snapshot", () => {
  const { container } = renderWithTheme(<Content />);
  expect(container.firstChild).toMatchSnapshot();
});
