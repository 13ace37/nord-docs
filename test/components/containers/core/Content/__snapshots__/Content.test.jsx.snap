// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
.c0 {
  width: 100%;
}

@media (min-width:75em) {
  .c0 {
    max-width: 75em;
  }
}

<div
  class="c0"
/>
`;
